### PR TITLE
Wave APIs

### PR DESCRIPTION
**Pull request by thorsteinnth**
Adds APIs for Wave clusters and Spark applications.

### CODE REVIEW

File: service/wave/wave.go [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	UpdatedAt *time.Time `json:"updatedAt,omitempty"`
+}
+
+type Environment struct {
+	OperatorVersion         *string `json:"operatorVersion,omitempty"`
+	CertManagerDeployed     *bool   `json:"certManagerDeployed,omitempty"`
+	K8sClusterProvisioned   *bool   `json:"k8sClusterProvisioned,omitempty"`
+	OceanClusterProvisioned *bool   `json:"oceanClusterProvisioned,omitempty"`
+	EnvironmentNamespace    *string `json:"environmentNamespace,omitempty"`
+	OceanClusterId          *string `json:"oceanClusterId,omitempty"`
```
Comment (liranp) [inline service/wave/wave.go:35]:
```suggestion
	OceanClusterID          *string `json:"oceanClusterId,omitempty"`
```
[Ref.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

File: service/wave/wave.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+}
+
+type Config struct {
+	Components []*Component `json:"components,omitempty"`
+}
+
+type Component struct {
+	Uid             *string           `json:"uid,omitempty"`
```
Comment (liranp) [inline service/wave/wave.go:43]:
```suggestion
	UID             *string           `json:"uid,omitempty"`
```
[Ref.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

File: service/wave/wave.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+	ShouldDeleteOcean *bool   `json:"shouldDeleteOcean,omitempty"`
+}
+
+type DeleteClusterOutput struct{}
+
+type ListSparkApplicationsInput struct {
+	ClusterIdentifier *string `json:"clusterIdentifier,omitempty"`
+	Name              *string `json:"name,omitempty"`
+	Namespace         *string `json:"namespace,omitempty"`
+	ApplicationId     *string `json:"applicationId,omitempty"`
```
Comment (liranp) [inline service/wave/wave.go:93]:
```suggestion
	ApplicationID     *string `json:"applicationId,omitempty"`
```
[Ref.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

File: service/wave/wave.go [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+	if input != nil {
+		if input.ClusterIdentifier != nil {
+			r.Params.Set("clusterIdentifier", spotinst.StringValue(input.ClusterIdentifier))
+		}
+
+		if input.Name != nil {
+			r.Params.Set("name", spotinst.StringValue(input.Name))
+		}
+
+		if input.ApplicationId != nil {
```
Comment (liranp) [inline service/wave/wave.go:209]:
```suggestion
		if input.ApplicationID != nil {
```
[Ref.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

File: service/wave/wave.go [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+			r.Params.Set("applicationId", spotinst.StringValue(input.ApplicationId))
```
Comment (liranp) [inline service/wave/wave.go:210]:
```suggestion
			r.Params.Set("applicationId", spotinst.StringValue(input.ApplicationID))
```
[Ref.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

**Review by liranp — CHANGES_REQUESTED** [review]
@thorsteinnth Could you please review my suggestions? Thanks!

**Comment by thorsteinnth** [comment]
> @thorsteinnth Could you please review my suggestions? Thanks!

Thanks @liranp - I have pushed fixes for your suggestions

**Review by liranp — APPROVED** [review]

**Comment by liranp** [comment]
Released [v1.80.0](https://github.com/spotinst/spotinst-sdk-go/releases/tag/v1.80.0).

**Pull request merged**